src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.Types.cs

```diff
@@ -0,0 +1,477 @@
+        Assert.Throws<NotSupportedException>(() => CallStaticMethod3(null));
+
+        [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "MethodName")]
+        extern static ref int CallStaticMethod1([UnsafeAccessorType(null!)] object a);
+
+        [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "MethodName")]
+        extern static ref int CallStaticMethod2([UnsafeAccessorType("_DoesNotExist_")] object a);
+
+        [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "MethodName")]
+        extern static ref int CallStaticMethod3([UnsafeAccessorType("S1")] object a);
+    }
+
+    [Fact]
+    public static void Verify_Type_StaticClass()
+    {
+        Console.WriteLine($"Running {nameof(Verify_Type_StaticClass)}");
+
+        var f = GetStaticClassField(null);
+        Assert.Equal(StaticClass.StaticField, f);
+        Assert.Equal(StaticClass.StaticField, CallStaticClassMethod(null));
+
+        [UnsafeAccessor(UnsafeAccessorKind.StaticField, Name = "StaticField")]
+        extern static ref int GetStaticClassField([UnsafeAccessorType("StaticClass")] object a);
+
+        [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "StaticMethod")]
+        extern static int CallStaticClassMethod([UnsafeAccessorType("StaticClass")] object a);
+    }
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    extern static TargetClass CreateTargetClass([UnsafeAccessorType("C2")] object a);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    extern static TargetClass CreateTargetClass([UnsafeAccessorType("C2&")] ref object a);
+
+    // Skip validating error cases on Mono runtime
+    [ConditionalFact(typeof(TestLibrary.Utilities), nameof(TestLibrary.Utilities.IsNotMonoRuntime))]
+    public static void Verify_Type_TypeCheck()
+    {
+        Console.WriteLine($"Running {nameof(Verify_Type_TypeCheck)}");
+
+        Assert.Throws<InvalidCastException>(() => CreateTargetClass(new C1()));
+        Assert.Throws<InvalidCastException>(() =>
+        {
+            object c1 = new C1();
+            CreateTargetClass(ref c1);
+        });
+    }
+
+    [Fact]
+    public static void Verify_Type_CallInstanceMethods()
+    {
+        Console.WriteLine($"Running {nameof(Verify_Type_CallInstanceMethods)}");
+
+        C2 c2 = new();
+        object arg = c2;
+        TargetClass tgt = CreateTargetClass(arg);
+
+        arg = new C1();
+        Assert.Equal(c2, CallM_C1(tgt, arg));
+        Assert.Equal(c2, CallM_RC1(tgt, ref arg));
+        Assert.Equal(c2, CallM_RROC1(tgt, ref arg));
+        Assert.Equal(c2, CallM_C1_RC2(tgt, arg));
+
+        object ic = null;
+        object rc = null;
+        object oc = null;
+        CallM_ByRefs(tgt, arg, in ic, ref rc, out oc);
+        Assert.Null(ic);
+        Assert.Equal(arg, rc);
+        Assert.Equal(arg, oc);
+
+        Assert.Equal(typeof(C1[]), CallM_C1Array(tgt, Array.Empty<C1>()));
+        Assert.Equal(typeof(C1[][]), CallM_C1MDArray2(tgt, new C1[0][]));
```

Comment:
This is jagged array.

MD arrays would be `C1[,]` or `C1[,,]`.

We need test coverage for both.

Reply:
Oh right. I always forget that.